spec/models/manageiq/providers/azure/cloud_manager_spec.rb

```diff
@@ -76,6 +76,53 @@
         @e.connect
       end
 
+      it "with http_proxy_uri" do
+        proxy = 'http://www.foo.bar'
+        allow(@e).to receive(:http_proxy_uri).and_return(proxy)
+
+        expect(described_class).to receive(:raw_connect) do |clientid, clientkey, azure_tenant_id, subscription, http_proxy_uri|
+          expect(clientid).to eq("klmnopqrst")
+          expect(clientkey).to eq("1234567890")
+          expect(azure_tenant_id).to eq("abcdefghij")
+          expect(subscription).to eq("fghij67890")
+          expect(http_proxy_uri).to eq(proxy)
+        end
+        @e.subscription = "fghij67890"
+        @e.connect
+      end
+
+      it "with provider_region" do
+        expect(described_class).to receive(:raw_connect) do |clientid, clientkey, azure_tenant_id, subscription, http_proxy_uri, provider_region|
+          expect(clientid).to eq("klmnopqrst")
+          expect(clientkey).to eq("1234567890")
+          expect(azure_tenant_id).to eq("abcdefghij")
+          expect(subscription).to eq("fghij67890")
+          expect(http_proxy_uri).to eq(nil)
+          expect(provider_region).to eq("westus2")
+        end
+
+        @e.subscription = "fghij67890"
+        @e.provider_region = "westus2"
+        @e.connect
+      end
+
+      it "with endpoint" do
+        expect(described_class).to receive(:raw_connect) do |clientid, clientkey, azure_tenant_id, subscription, http_proxy_uri, provider_region, endpoint|
+          expect(clientid).to eq("klmnopqrst")
+          expect(clientkey).to eq("1234567890")
+          expect(azure_tenant_id).to eq("abcdefghij")
+          expect(subscription).to eq("fghij67890")
```

Comment:
@djberg96 why are lines 111  through 114  repeated so many times in this spec?

Reply:
@bronaghs Just to make sure they weren't negatively affected. I can of course remove them if you feel they're redundant.